Cython/Compiler/ExprNodes.py

```diff
@@ -13563,7 +13558,81 @@ def generate_result_code(self, code):
             code.error_goto_if_null(self.result(), self.pos)))
         code.put_gotref(self.result())
 
+class AnnotationNode(ExprNode):
+    # Deals with the two possible uses of an annotation.
+    # 1. The post PEP-563 use where an annotation is stored
+    #  as a string
+    # 2. The Cython use where the annotation can indicate an
+    #  object type
+    #
+    # doesn't handle the pre PEP-563 version where the
+    # annotation is evaluated into a Python Object
+
+    subexprs = []
+    def __init__(self, pos, expr, string=None):
+        """string is expected to already be a StringNode or None"""
+        ExprNode.__init__(self, pos)
+        if string is None:
+            # import doesn't work at top of file?
+            from .AutoDocTransforms import AnnotationWriter
+            string = StringEncoding.EncodedString(AnnotationWriter().write(expr))
```

Comment:
I've added a warning for these. I originally tried looking for ways to just copy the original code rather than regenerating it, but I couldn't work out a good way to do it. It's the sort of change that would be easy to make in the future though.